### PR TITLE
Synced in out audio, functions WavLen<>, WavPos<>, WavRem<>, Pct<>, Mult<>

### PR DESCRIPTION
**Pull request by artandtechvof**
This is the first attempt of merging the new functions that allow the wav-lentgh to be used a a paramater for bladestyle effects as discussed and demonstrated on the rebelarmory.

There is a warning that these will not merge automatically, so I want to start checking what is needed to get this into the main fork (if at all?) 
Also start sparring with you to see if you want to alter the way this feature is implemented.

Currently there is no coupling between sound effect and blade effect directly. This is all distributed by saber_base.h, When an effect is activated, balde and- sound classes do there own thing. The blade style does not know directly which soundfile is selected and which buffered_wavplayer is playing this sound.
My solution was to introduce an extra item to the wav_player 'effectname' and '2nd effectname'. (because pstoff is added as second effect to the wavplayer, and therefore is not standalone like preon)
The WavLen functions have a look-up table to cross-correlate EFFECT_NAME and SFX# effect names. Using the soundeffect name and blade effect name, we can now cycle trough the wavplayers (there are only a couple of them) to find the one that is playing the effect we are after, and thus being able to track its length, current position and remaining time.

I have been looking into using the OneshotEffectDetector to figure out which effect is active, but was unable to retrieve the wav_player that was used for the effect. 

The helper functions Pct<> and Mult<> were separated from the WavLen functions as you suggested, and can be re-used for any purpose within the OS architecture.

Generic wav timer alterations to the code;   
  playwav   -> debug and effectname()
  bufferedwav  -> effectname()
  effect.h  name_ -> public
  blade_base.h -> added EFFECT_PSTOFF
  Optional  ifon -> debug info
added;
  functions/wav_time
  functions/wav_effect
  functions/mult
  functions/pct
  ProffieOS.ino -> include wav_time.time, pct.h, mult.h

### CODE REVIEW

File: ProffieOS.ino [hunk 1]
```diff
@@ -29,10 +29,12 @@
 // #define CONFIG_FILE "config/owk_v2_config.h"
 // #define CONFIG_FILE "config/test_bench_config.h"
 // #define CONFIG_FILE "config/toy_saber_config.h"
-#define CONFIG_FILE "config/proffieboard_v1_test_bench_config.h"
+//#define CONFIG_FILE "config/proffieboard_v1_test_bench_config.h"
 // #define CONFIG_FILE "config/td_proffieboard_config.h"
 // #define CONFIG_FILE "config/teensy_audio_shield_micom.h"
 // #define CONFIG_FILE "config/proffieboard_v2_ob4.h"
+#define CONFIG_FILE "config/aat_proffie_singleblade_config.h"
```
Comment (profezzorn) [inline ProffieOS.ino:36]: undo

File: ProffieOS.ino [hunk 2]
```diff
@@ -136,6 +138,7 @@
 
 #include <i2c_t3.h>
 #include <SD.h>
+//#include <SPI.h>
```
Comment (profezzorn) [inline ProffieOS.ino:141]: undo

File: ProffieOS.ino [hunk 3]
```diff
@@ -782,9 +786,9 @@ class Commands : public CommandParser {
     if (!strcmp(cmd, "sdtest")) {
       SDTestHelper sdtester;
       if (e && !strcmp(e, "all")) {
-	sdtester.TestDir("");
+  sdtester.TestDir("");
```
Comment (profezzorn) [inline ProffieOS.ino:789]: undo

File: ProffieOS.ino [hunk 4]
```diff
@@ -782,9 +786,9 @@ class Commands : public CommandParser {
...
       } else {
-	sdtester.TestFont();
+  sdtester.TestFont();
```
Comment (profezzorn) [inline ProffieOS.ino:791]: undo

**Review by profezzorn — COMMENTED** [review]
There is a lot of unrelated changes in this CL, please sync, undo anything that isn't related to this change and let me know when to proceed with reviewing.

**Comment by artandtechvof** [comment]
I will close this one, and do a new pull request from a cleaned up branch

**Pull request closed without merging**